### PR TITLE
MF-641: Completed Patient List Modal

### PR DESCRIPTION
**Pull request by vasharma05**
### What does this PR include?

This PR contains the final completion changes to the patient list modal.

image

### CODE REVIEW

File: packages/esm-patient-list-app/src/AddPatientToList/index.tsx [hunk 1]
```diff
@@ -2,76 +2,127 @@ import React, { useState, useEffect, useMemo, useCallback } from 'react';
...
-import { toOmrsIsoString } from '@openmrs/esm-framework';
+import { toOmrsIsoString, showToast, showModal, usePagination } from '@openmrs/esm-framework';
 import Search from 'carbon-components-react/lib/components/Search';
 import Button from 'carbon-components-react/lib/components/Button';
+import Pagination from 'carbon-components-react/lib/components/Pagination';
 import Checkbox from 'carbon-components-react/lib/components/Checkbox';
 import SkeletonText from 'carbon-components-react/es/components/SkeletonText';
 import styles from './add-patient-to-list.scss';
 
+let closeModal = () => {};
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/AddPatientToList/index.tsx:13]: Please don't make it global.

File: packages/esm-patient-list-app/src/AddPatientToList/index.tsx [hunk 2]
```diff
@@ -2,76 +2,127 @@ import React, { useState, useEffect, useMemo, useCallback } from 'react';
...
+
+export const openModal = () => {
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/AddPatientToList/index.tsx:15]: Why is this exported? Why don't you just use the `showModal` in the hook?

**Comment by vasharma05** [comment]
https://user-images.githubusercontent.com/51502471/127122492-d87b3daa-0dc7-4693-bccd-aad2f1187653.mp4

@FlorianRappl @jonathandick I have tried to show the modal and the workflow too. @FlorianRappl this video also contains the workflow, which I want you to go through, it will help in solving out the queries you pointed above and can you please suggest me something better, if this workflow doesn't suite you.

@jonathandick I want you to please check the text shown in the toasts and direct if any changes are required.

File: packages/esm-patient-list-app/src/AddPatientToList/index.tsx [hunk 3]
```diff
@@ -2,76 +2,127 @@ import React, { useState, useEffect, useMemo, useCallback } from 'react';
...
+};
+
+function getPatientUuidFromUrl(): string {
+  const match = /\/patient\/([a-zA-Z0-9\-]+)\/?/.exec(location.pathname);
+  return match && match[1];
+}
+
 interface AddPatientProps {
   close: () => void;
   patientUuid: string;
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/AddPatientToList/index.tsx:26]: Not used.

**Review by FlorianRappl — COMMENTED** [review]
I already suggested something better.

1. Remove the global `closeModal`.
2. Remove the `openModal`.
3. Call `showModal` directly in the `AddPastVisitOverflowMenuItem` component:

```jsx
const openModal = React.useCallback(() => {
  const dispose = showModal('add-patient-to-patient-list-modal', {
    close: () => dispose();
  });
}, []);
```

File: packages/esm-patient-list-app/src/AddPatientToList/index.tsx [hunk 4]
```diff
@@ -2,76 +2,127 @@ import React, { useState, useEffect, useMemo, useCallback } from 'react';
...
-const AddPatient: React.FC<AddPatientProps> = ({ close, patientUuid }) => {
+interface PatientListProp {
+  name: string;
+  visible: boolean;
+  selected: boolean;
+}
+
+type PatientListObj = Record<string, PatientListProp>;
+
+const AddPatient: React.FC<AddPatientProps> = ({ close }) => {
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/AddPatientToList/index.tsx:37]: The `close` seems to be unused, too.

Remove the prop.

**Comment by vasharma05** [comment]
Thanks @FlorianRappl ! I didn't think of using it this way.
Thanks alot!

**Review by FlorianRappl — APPROVED** [review]
LGTM

**Pull request merged**